staging/src/k8s.io/kubectl/pkg/util/openapi/openapi_getter.go

```diff
@@ -19,47 +19,36 @@ package openapi
 import (
 	"sync"
 
+	openapi_v2 "github.com/googleapis/gnostic/openapiv2"
 	"k8s.io/client-go/discovery"
 )
 
 // synchronizedOpenAPIGetter fetches the openapi schema once and then caches it in memory
 type synchronizedOpenAPIGetter struct {
+	openAPIClient discovery.OpenAPISchemaInterface
+
 	// Cached results
 	sync.Once
-	openAPISchema Resources
+	openAPISchema *openapi_v2.Document
 	err           error
-
-	openAPIClient discovery.OpenAPISchemaInterface
 }
 
-var _ Getter = &synchronizedOpenAPIGetter{}
-
-// Getter is an interface for fetching openapi specs and parsing them into an Resources struct
-type Getter interface {
-	// OpenAPIData returns the parsed OpenAPIData
-	Get() (Resources, error)
-}
+var _ discovery.OpenAPISchemaInterface = &synchronizedOpenAPIGetter{}
 
 // NewOpenAPIGetter returns an object to return OpenAPIDatas which reads
 // from a server, and then stores in memory for subsequent invocations
-func NewOpenAPIGetter(openAPIClient discovery.OpenAPISchemaInterface) Getter {
+func NewOpenAPIGetter(openAPIClient discovery.OpenAPISchemaInterface) *synchronizedOpenAPIGetter {
```

Comment:
Shouldn't you return `discovery.OpenAPISchemaInterface`? Can you even return a private type?

Reply:
Oh yeah sounds like I shouldn't do this

Reply:
Fixed.